compiler/rustc_resolve/src/ident.rs

```diff
@@ -1027,15 +1027,21 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             if !self.is_accessible_from(import_vis, parent_scope.module) {
                 continue;
             }
-            let module = match glob_import.imported_module.get() {
+            let imported_module = match glob_import.imported_module.get() {
                 Some(ModuleOrUniformRoot::Module(module)) => module,
                 Some(_) => continue,
                 None => return Err((Undetermined, Weak::Yes)),
             };
+
+            if ptr::eq(imported_module, module) {
```

Comment:
The "cannot glob-import a module into itself" error is reported for the case `glob_import.imported_module == glob_import.parent_scope.module`.
I'm not sure `module` here is always the same as `glob_import.parent_scope.module`, it needs at least an assert.

Or just check this condition on `glob_import.parent_scope.module` directly.